src/seedu/addressbook/data/person/Address.java

```diff
@@ -34,25 +46,131 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
     public static boolean isValidAddress(String test) {
         return test.matches(ADDRESS_VALIDATION_REGEX);
     }
-
-    @Override
-    public String toString() {
-        return value;
+    
+    public Block getBlock(){
+    	return block;
+    }
+    
+    public Street getStreet(){
+    	return street;
+    }
+    
+    public Unit getUnit(){
+    	return unit;
+    }
+    
+    public Postal getPostal(){
+    	return postal;
+    }
+    
+    public boolean matchingBlock(Block block1){
+       return block.getBlockName().equals(block1.getBlockName());
     }
+    
+    public boolean matchingStreet(Street street1){
+        return street.getStreetName().equals(street1.getStreetName());
+     }
+    
+    public boolean matchingUnit(Unit unit1){
+        return unit.getUnitName().equals(unit1.getUnitName());
+     }
+    
+    public boolean matchingPostal(Postal postal1){
+    	return postal.getPostalDigit().equals(postal1.getPostalDigit());
+    }
+    
 
     @Override
     public boolean equals(Object other) {
-        return other == this // short circuit if same object
-                || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
+        if(other instanceof Address){
+        	Address otherAddress = (Address) other;
+        	return matchingBlock(otherAddress.getBlock()) && matchingStreet(otherAddress.getStreet()) && matchingUnit(otherAddress.getUnit()) && matchingPostal(otherAddress.getPostal());
+        }
+        return false;
     }
 
     @Override
+    public String toString() {
+        return value;
+    }
+    
+    @Override
     public int hashCode() {
         return value.hashCode();
     }
 
     public boolean isPrivate() {
         return isPrivate;
     }
+    
+    
+    public class Block{
+    	private String _blockName;
```

Comment:
This naming style (using `_` as prefix/suffix for local variables), although allowed by our coding standard, is not consistent with the rest of the code base.